charset_normalizer/utils.py

```diff
@@ -197,6 +198,17 @@ def is_unicode_range_secondary(range_name: str) -> bool:
     return any(keyword in range_name for keyword in UNICODE_SECONDARY_RANGE_KEYWORD)
 
 
+@lru_cache(maxsize=UTF8_MAXIMAL_ALLOCATION)
+def is_unprintable(character: str) -> bool:
+    if (
```

Comment:
Why not simply inline directly with a return statement?

Reply:
The function is cached, and there is almost no difference in performance. I leave the choice to you.
```python3
Total time: 0.010625 s                                                                                
File: /home/user/Downloads/charset_normalizer-master/charset_normalizer/utils.py                      
Function: is_unprintable at line 201

Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
   201                                           @lru_cache(maxsize=UTF8_MAXIMAL_ALLOCATION)
   202                                           @profile
   203                                           def is_unprintable(character: str) -> bool:
   204     14995       3538.0      0.2     33.3      if (
   205      7354       3100.0      0.4     29.2          character.isspace() is False  # includes \n \t \r \v
   206      7338       2261.0      0.3     21.3          and character.isprintable() is False
   207       303         78.0      0.3      0.7          and character != "\x1A"  # Why? Its the ASCII substitute character.
   208                                               ):
   209       302         67.0      0.2      0.6          return True
   210      7052       1581.0      0.2     14.9      return False
```
```python3
Total time: 0.008757 s
File: /home/user/Downloads/charset_normalizer-master/charset_normalizer/utils.py
Function: is_unprintable at line 201

Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
   201                                           @lru_cache(maxsize=UTF8_MAXIMAL_ALLOCATION)
   202                                           @profile
   203                                           def is_unprintable(character: str) -> bool | None:
   204      7354       1456.0      0.2     16.6      return (
   205     14692       4777.0      0.3     54.6          character.isspace() is False  # includes \n \t \r \v
   206      7338       2442.0      0.3     27.9          and character.isprintable() is False
   207       303         82.0      0.3      0.9          and character != "\x1A"  # Why? Its the ASCII substitute character.
   208                                               )
```
```python3
Total time: 0.011836 s
File: /home/user/Downloads/charset_normalizer-master/charset_normalizer/utils.py
Function: is_unprintable at line 201

Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
   201                                           @lru_cache(maxsize=UTF8_MAXIMAL_ALLOCATION)
   202                                           @profile
   203                                           def is_unprintable(character: str) -> bool:
   204     22047       5027.0      0.2     42.5      return True if (
   205      7354       3031.0      0.4     25.6          character.isspace() is False  # includes \n \t \r \v
   206      7338       2245.0      0.3     19.0          and character.isprintable() is False
   207       303         74.0      0.2      0.6          and character != "\x1A"  # Why? Its the ASCII substitute character.
   208      7052       1459.0      0.2     12.3      ) else False
```

Reply:
```python3
Total time: 0.010625 s                                                                                
File: /charset_normalizer-master/charset_normalizer/utils.py                      
Function: is_unprintable at line 201

Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
   201                                           @lru_cache(maxsize=UTF8_MAXIMAL_ALLOCATION)
   202                                           @profile
   203                                           def is_unprintable(character: str) -> bool:
   204     14995       3538.0      0.2     33.3      if (
   205      7354       3100.0      0.4     29.2          character.isspace() is False  # includes \n \t \r \v
   206      7338       2261.0      0.3     21.3          and character.isprintable() is False
   207       303         78.0      0.3      0.7          and character != "\x1A"  # Why? Its the ASCII substitute character.
   208                                               ):
   209       302         67.0      0.2      0.6          return True
   210      7052       1581.0      0.2     14.9      return False

```
```python3
Total time: 0.008766 s
File: /charset_normalizer-master/charset_normalizer/utils.py
Function: is_unprintable at line 201

Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
   201                                           @lru_cache(maxsize=UTF8_MAXIMAL_ALLOCATION)
   202                                           @profile
   203                                           def is_unprintable(character: str) -> bool:
   204      7354       1462.0      0.2     16.7      return (
   205     14692       4697.0      0.3     53.6          character.isspace() is False  # includes \n \t \r \v
   206      7338       2530.0      0.3     28.9          and character.isprintable() is False
   207       303         77.0      0.3      0.9          and character != "\x1A"  # Why? Its the ASCII substitute character.
   208                                               )
```

Reply:
In favor of the 2nd.

Reply:
Done